apps/labs/posts/numba-polynomial-support.md

```diff
@@ -0,0 +1,173 @@
+    <div class="w-1/2">
+      <img alt="Numba logo."
+          src="/posts/numba-polynomial-support/numba_logo.png" 
+          class="h-[8em] lg:h-1/2 mx-auto lg:mt-4" />
+    </div>
+    <div class="w-1/2">
+      <img alt="Quansight Labs logo."
+          src="/posts/numba-polynomial-support/qlabs_logo.png" 
+          class="h-[8em] lg:h-1/2 mx-auto lg:mt-4" />
+    </div>
+  </div>
+</div>
+
+## What is Numba?
+
+[Numba](https://numba.pydata.org/) is a just-in-time compiler for Python that
+translate a subset of Python and NumPy into fast machine code. The most common
+way to use Numba is through its collection of decorators that can be applied to
+your functions to instruct Numba to compile them.
+
+When a call is made to a function using the `@jit` decorator from Numba, it is
+compiled to machine code “just-in-time” for execution and all or part of your
+code can subsequently run at native machine code speed! Numba reads the Python
+bytecode for a decorated function and combines this with information about the
+types of the input arguments to the function. It analyzes and optimizes your
+code, and finally uses the [LLVM compiler infrastructure](https://llvm.org/) to
+generate a machine code version of your function, tailored to your CPU
+capabilities. This compiled version is then used when your function is called.
+
+The Numba `@jit` decorator fundamentally operates in two compilation modes,
+`nopython` mode and `object` mode. The behaviour of the `nopython` compilation
+mode is to essentially compile the decorated function so that it will run
+entirely without the involvement of the Python interpreter. This is the
+recommended and best-practice way to use the Numba jit decorator as it leads to
+the best performance. It is important to note that nopython mode is not
+guaranteed to compile successfully. This can happen if there are unsupported
+functions or classes which are used in the scope of the `@jit` decorator.
+
+<center>
+  <img
+    alt="Numba schematic showing different stages of compilation."
+    src="/posts/numba-polynomial-support/numba-schematic.png"
+    class="mb-0 mt-16"
+    />
+  <br/>
+  <i>How Numba works. Taken from <a href="https://towardsdatascience.com numba-weapon-of-mass-optimization-43cdeb76c7da">here</a>.</i>
+</center>
+
+## Increasing support for NumPy functions in Numba
+
+One of the current priorities of Numba is having seamless integration with
+NumPy. As an intern at Quansight Labs, my main task was extending the list of
+NumPy functions that are supported by Numba. Since Numba does not understand
+calls to NumPy functions, developers have to reimplement such functions. The
+easiest way to do so is using the `@numba.extending.overload` decorator, which
+lets the developer write a new implementation of the function in Python. The new
+implementation needs to have the same behaviour as the original function, but it
+should also be completely supported by Numba, e.g. it cannot call NumPy
```

Comment:
Depending on the specific intent for the meaning here, "i.e.," might be the better choice.

It depends on whether "it cannot call NumPy functions which are not supported yet" is the _single_ reason why a reimplementation would not be suitable (we'd use 'i.e.'), or one of multiple reasons why it would not be suitable (we'd use 'e.g.').

```suggestion
should also be completely supported by Numba; e.g., it cannot call NumPy
```

Reply:
there's probably more reasons - accepting the suggestion.